codec/encode.go

```diff
@@ -1176,8 +1176,8 @@ func (e *Encoder) deferred(err1 *error) {
 	e.alwaysAtEnd()
 	if recoverPanicToErr {
 		if x := recover(); x != nil {
-			panicValToErr(e, x, err1)
-			panicValToErr(e, x, &e.err)
+			*err1 = panicValToErr(e, x)
+			e.err = *err1
```

Comment:
Seems better here to avoid calling panicValToErr (and the reflection calls) twice.